execution-engine/src/native_modules/postcard.rs

```diff
@@ -0,0 +1,83 @@
+use crate::fs::{FileSystem, FileSystemResult, Service};
+use postcard::from_bytes;
```

Comment:
The native module is part of the runtime. For this demonstration, we deserialise the postcard encoding and serializing to JSON string. Postcard encoding requires the type in compilation time rather than dynamically detecting the type in runtime.

I will expect this demonstration to be removed and replaced by more realistic encryption and decryption.